model/event/subscriptions.go

```diff
@@ -824,6 +825,14 @@ func NewSpawnhostExpirationSubscription(owner string, sub Subscriber) Subscripti
 	return NewSubscriptionByOwner(owner, sub, ResourceTypeHost, TriggerExpiration)
 }
 
+// NewSpawnHostIdleWarningSubscription returns a subscription for the spawn host.
+func NewSpawnHostIdleWarningSubscription(hostId string, sub Subscriber) Subscription {
+	subscription := NewSubscriptionByID(ResourceTypeHost, TriggerSpawnHostIdle, hostId, sub)
+	// Assign hostID as the subscription to avoid having multiple idle subscriptions on the same host.
+	subscription.ID = hostId
```

Comment:
Setting a subscription ID to a host ID for this niche situation may become a bit confusing in the future -- can we add something to the ID to make it clear that this subscription is specficially for sending idle warnings? I.e. `fmt.Sprintf("idle-warning-%s", hostId)`?